[feat] 도메인 설계

GGHDMS opened the pull request:
This closes #1

GGHDMS committed:
feat: AuditingFileds 구현

GGHDMS committed:
feat: JPA Auditing 기능 적용을 위한 config 작성

GGHDMS committed:
feat: ERD 설계에 맞는 Enum 타입 작성

GGHDMS committed:
feat: ERD 설계에 맞는 Entity 클래스 작성

hxnsxxm commented:
엔티티에 테이블을 매핑하는 코드가 추가되어야 할 듯 합니다.
(1) @Entity(name="name")
(2) @Table(name="name")
두 가지 중의 한 방법으로요

hxnsxxm commented:
유저 항목에 있는 technology 부분이 제가 설계를 잘못한 거 같아서 이 부분은 잠시 보류하겠습니다

GGHDMS committed:
feat: 코드 리팩토링

`User` 와 `Technology` 는 @ManytoMany 관계이므로 중간에 테이블을 하나 더 추가
`Location` 에 컬럼명 설정 빠져 있어서 추가

GGHDMS committed:
refactor: 단순 코드 띄어쓰기 및 공백 제거

GGHDMS committed:
refactor: jpa properties 수정

koseha reviewed (approved):
👍  수고하셨습니다
